src/main/java/frc/robot/subsystems/ShooterSubsystem.java

```diff
@@ -0,0 +1,107 @@
+package frc.robot.subsystems;
+
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+
+import com.ctre.phoenix.motorcontrol.ControlMode;
+import com.ctre.phoenix.motorcontrol.FeedbackDevice;
+import com.ctre.phoenix.motorcontrol.can.WPI_TalonSRX;
+import frc.robot.config.Config;
+
+public class ShooterSubsystem extends SubsystemBase {
+
+  // Singleton instance for ShooterSubsystem
+  private static final ShooterSubsystem INSTANCE_SHOOTER = new ShooterSubsystem();
+
+  private WPI_TalonSRX m_shooter;
+
+  // Indicates use of the primary PID loop rather than cascaded ones
+  int kPIDLoopIDX = 0;
+
+	 // Set to zero to skip waiting for confirmation, set to nonzero to wait and
+	 // report to DS if action fails.
+  int kTimeoutMs = 30;
+
+  // Protobot PID values
+  double kF = 0.037;
+  double kP = 0.5; 
+  double kI = 0;
+  double kD = 10;
+
+  double velocityModeUnits;
+
+  private ShooterSubsystem() {
+    // Initialize a private variable for the motor
+    if (Config.SHOOTER_MOTOR != -1){
+      m_shooter = new WPI_TalonSRX(Config.SHOOTER_MOTOR);
+    }
+
+    // Factory Default to prevent unexpected behaviour
+    m_shooter.configFactoryDefault();
+
+    // Config sensor used for PID control
+    m_shooter.configSelectedFeedbackSensor(FeedbackDevice.CTRE_MagEncoder_Relative, 
+    kPIDLoopIDX, kTimeoutMs); 
```

Comment:
eventually add error checking here in case the encoder isn't connected, we don't want to try running a closed loop mode